Filings/Reports panel amendment information

xtine opened the pull request:
This adds amendment information to the modal panel on filings page.

<img width="996" alt="screen shot 2017-01-17 at 6 56 56 pm" src="https://cloud.githubusercontent.com/assets/24054/22047940/4f060418-dce7-11e6-9423-192b65a4e4fd.png">

Relies on: https://github.com/18F/fec-style/pull/615

https://github.com/18F/openFEC-web-app/issues/1738

xtine committed:
amendment panel

xtine committed:
amendment version

xtine committed:
merge

xtine committed:
amendment version info in helper functions

xtine committed:
amendment panel updates

xtine committed:
merge fix

xtine committed:
Merge branch 'develop' into feature/amendment-panel

xtine committed:
version unknown

noahmanger commented:
@xtine is this still in progress or should I review and merge?

xtine commented:
It's still in progress. There's still some minor things to be added, but also waiting for OK that there's cleaner amendment status data.

xtine committed:
Merge branch 'develop' into feature/amendment-panel

xtine committed:
version unknown space

xtine committed:
Merge branch 'develop' into feature/amendment-panel

xtine committed:
add filters to filings

xtine committed:
reports filter

xtine committed:
reports pages document status

xtine committed:
remove past version option

dogweather commented on static/js/modules/helpers.js:
```diff
@@ -216,6 +215,41 @@ function zeroPad(container, item, appendee) {
   });
 }
 
+function amendmentVersion(data) {
```
Could there be some variable name that'd be more expressive than `data`? It seems like a flag for whether the version is the current one. So maybe, `isCurrent`?

xtine replied:
I changed the variable to the more descriptive (and what the parameter is actually returned from API) `most_recent`.

dogweather commented on static/js/modules/helpers.js:
```diff
@@ -216,6 +215,41 @@ function zeroPad(container, item, appendee) {
+
+  // because of messy data, do not show if not e-filing or null amendment indicator
+  if (row.means_filed === 'e-file' && row.amendment_indicator !== null) {
+    if (row.amendment_indicator === 'N') {
```
It'd be great to see the magic value `'N'` refactored from here.

dogweather replied:
Also, the magic string `'e-file'`.

xtine replied:
`N` and `e-file` are variables returned from the API. How would I refactor that?

dogweather replied:
I think probably with constants. Roughly like,

```javascript
Api = {}
Api.MEANS_E_FILED = 'e-file'
Api.AMENDMENT_INDICATOR_NEW = 'N'
```

This way, the API is defined in just one place and misspellings are more likely to get caught:

```javascript
if (row.means_filed === Api.MEANS_E_FILED)
```

dogweather commented on static/js/modules/helpers.js:
```diff
@@ -216,6 +215,41 @@ function zeroPad(container, item, appendee) {
+  var description = '';
+  var amendment_num = 1;
+
+  // because of messy data, do not show if not e-filing or null amendment indicator
```
IMO an expressive guard clause would obviate the redundant comment, capture the described logic more clearly, and result in less nesting. E.g.,

```javascript
// Don't show if we don't have the appropriate data
if (row.means_filed !== 'e-file' || row.amendment_indicator === null {
  return '',
}
```

xtine replied:
Thanks for the suggestion! I didn't want to nest much but didn't think of a cleaner way at the moment.

xtine committed:
update amendment version helper variable

xtine committed:
remove nesting in amendment indicator helper

noahmanger commented on openfecwebapp/templates/partials/filings-filter.html:
```diff
@@ -37,7 +38,33 @@
   <div class="accordion__content">
-    {% include 'partials/filters/amendment-indicator.html' %}
+    <div class="filter">
+      <fieldset class="js-filter js-filter-control" data-filter="checkbox" data-modifies-filter="most_recent" data-modifies-property="dataset">
```
Do you need the `js-filter-control` class and `data-modifies-*` attributes? Those are used when a filter needs to prompt a change in another one, iirc.

xtine replied:
You're correct, I forgot to remove them.

noahmanger commented on openfecwebapp/templates/partials/reports-filter.html:
```diff
@@ -31,19 +31,33 @@
+        <legend class="label">Document status</legend>
+        <ul>
+          <li>
+            <input id="amendment_indicator_original" name="amendment_indicator" type="checkbox" value="N">
```
I noticed that this filter doesn't seem to be supported by the API. Should check with @jontours .

noahmanger commented on static/js/modules/filings.js:
```diff
@@ -34,6 +34,10 @@ var renderModal = tables.modalRenderFactory(
         response.results[0] :
         {};
+
+      result.amendment_version = helpers.amendmentVersion(result.most_recent);
```
I think you need this same thing on the reports pages. It's missing from the panels.

xtine replied:
Just added to panels on reports.

Commit:
Switching amendment indicator for version

@xtine I noticed that committee pages were throwing a datatables error
because it was looking for a column that didn’t exist. Making the
change was faster than explaining in a comment :)

noahmanger commented on static/js/pages/filings.js:
```diff
@@ -41,6 +41,8 @@ $(document).ready(function() {
     }
   });
 
+  $('.panel__navigation').hide();
```
Why hide instead of just removing from the templates?

xtine replied:
Because the template is generic and then will remove it from other pages.

noahmanger replied:
👍

noahmanger reviewed (changes requested):
Looking mostly good, functionality wise. Had a few questions and made a small commit fixing a bug on committee page filings tables.

xtine committed:
remove unneeded properties

xtine committed:
set API vars

xtine committed:
show amendment data in reports panel

xtine commented:
This is ready for another review to get into release. @noahmanger

noahmanger commented on static/templates/reports/pac.hbs:
```diff
@@ -5,9 +5,16 @@
       (This report has been amended)
```
Let's remove this part from all these templates. Redundant with the other new info.

noahmanger replied:
(Note, these are in the other panel templates as well).

xtine replied:
With the rollback on amendment info on reports, we'll have to keep this in for reports panels.

noahmanger reviewed (commented):
Looks good to me. Caught one small thing.

xtine committed:
roll back amendment info on reports

noahmanger commented:
Looks like @dogweather 's comments have been addressed but just want to give him a ping to approve if everything looks good.

dogweather commented:
Excellent! Yep, looks great.

noahmanger commented:
Bingo. Thanks!
